website/blog/corba.md

```diff
@@ -0,0 +1,138 @@
+have come and gone. It is [widely believed][why_corba_failed] that these
+frameworks failed because they attempt to make remote method calls **look** and
+**act** the same as local method calls. This flies in the face of the [fallacies
+of distributed computing][fallacies], right? There are [whole
+papers][a_note_on_dc] explaining that frameworks trying to unify local and
+remote objects are "doomed to failure". Does this mean that Service Weaver is
+doomed to fail just like CORBA?
+
+No.
+
+In this blog post, we explain that remote method calls in Service Weaver
+do **look** like local calls, but that's actually okay. They don't, however,
+**act** like local method calls, which is where a framework can get in trouble.
+We'll also explain that CORBA didn't actually fail for any these reasons and
+why Service Weaver is not "doomed" to the same fate.
+
+## Looking Local
+
+- When we say a remote method call **looks** like a local method call, we mean
+  they **syntactically** look the same. We discuss the semantics of the calls,
+  or the way they act, in the next section.
+- Typically, remote method calls are syntactically pronounced. To execute an RPC
+  with gRPC, for example, you first have to construct [an RPC
+  client](https://grpc.io/docs/languages/go/basics/#creating-a-stub), making it
+  clear that a call is about to happen over the network. Then, when you issue
+  the RPC, you have to [pass protobufs as
+  arguments](https://grpc.io/docs/languages/go/basics/#simple-rpc-1), again
+  making it clear that the arguments are being serialized and sent over the
+  network.
+- Some frameworks choose to add additional syntax to remote calls even when it
+  is otherwise not needed. Ray, for example, requires that a function `f` be
+  called as
+  [`f.remote`](https://docs.ray.io/en/latest/ray-core/walkthrough.html#running-a-task)
+  when invoked remotely.
+- However, placing a syntactic burden on developers to make remote method calls
+  is unnecessary. Even [the paper][a_note_on_dc] which describes why unifying
+  local and remote objects is doomed to fail does "not argue against pleasant
+  programming interfaces."
+- Some people argue that a syntactic marker for remote method calls is needed to
+  see that a remote call is happening. However, any syntactic marker is
+  inadequate, as a remote method call can be easily wrapped inside a function,
+  and callers of this function now don't have the syntactic marker to see a
+  remote method call is happening under the covers.
+- Service Weaver remote method calls are syntactically identical to local method
+  calls. They do **look** the same, and this is okay. TODO: Give a short
+  example.
+- NOTE(mwhittaker): There might be an argument that using futures makes it
+  syntactically obvious a remote call could be happening, and futures cannot
+  easily be erased by wrapping them in functions. I think it might be too fine
+  of a point to bring up?
```

Comment:
Futures are used for other things than remote calls:
- work done on another thread
- IO (even local)
- Sleeps / timers

And this in turn infects all the way up and pretty much everything turns into returning futures (or being marked async) and it again becomes hard to tell what is local and what is remote.

I don't think it is worth mentioning futures.